GoTrue/src/commonMain/kotlin/io/github/jan/supabase/gotrue/providers/builtin/IDToken.kt

```diff
@@ -0,0 +1,38 @@
+package io.github.jan.supabase.gotrue.providers.builtin
+
+import io.github.jan.supabase.exceptions.SupabaseEncodingException
+import io.github.jan.supabase.supabaseJson
+import kotlinx.datetime.Instant
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.MissingFieldException
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.decodeFromJsonElement
+
+/**
+ * Authentication method with id token, client id, provider and optionally nonce
+ */
+object IDToken : DefaultAuthProvider<IDToken.Config, IDToken.Result> {
+
+    @Serializable(with = DefaultAuthProvider.Config.Companion::class)
+    data class Config(var id_token: String = "", var client_id: String = "", var provider: String = "", var nonce: String? = null): DefaultAuthProvider.Config()
```

Comment:
Probably best to stick with camel case:

```suggestion
    data class Config(@SerialName("id_token") var idToken: String = "", @SerialName("client_id") var clientId: String = "", var provider: String = "", var nonce: String? = null): DefaultAuthProvider.Config()
```